service/history/task/task.go

```diff
@@ -447,6 +454,21 @@ func (t *taskImpl) shouldResubmitOnNack() bool {
 		(t.queueType == QueueTypeActiveTransfer || t.queueType == QueueTypeActiveTimer)
 }
 
+func (t *taskImpl) shouldDropTaskOnError() bool {
+	if t.shard.GetConfig().EnableDropStuckTaskByDomainID(t.Info.GetDomainID()) && // use domainID here to avoid accessing domainCache
+		t.GetAttempt() > t.criticalRetryCount()/2 {
```

Comment:
I like that you are adding the attempt count here to ensure we don't drop transfer tasks on the first occurrence of an ErrMissingWorkflowStartEvent, since it can happen due to some transient DB issue. But, why divide the critical retry count by 2 instead of using it as is?

Reply:
It doesn't have to be half of that value. It just needs to be smaller than that value to make sure we don't generate logs for the stuck tasks.